Rakefile

```diff
@@ -1,29 +1,124 @@
-require 'rspec/core/rake_task'
 require 'fileutils'
 require 'rototiller'
+require "yard"
+require "rubocop/rake_task"
+require "flog_task"
+require "flay_task"
+require "roodi_task"
+require "rubycritic/rake_task"
+
+YARD_DIR = "doc".freeze
+DOCS_DIR = "docs".freeze
 
 task :default do
-  sh %{rake -T}
+  sh %(rake -T)
+end
+
+namespace :gem do
+  require "bundler/gem_tasks"
 end
 
-# temporary backwards compat
-task :test => :'test:unit'
-task :acceptance => :'test:acceptance'
+task yard: :"docs:yard"
+
+# rubocop:disable Metrics/BlockLength
+namespace :docs do
+  # docs:yard task
+  YARD::Rake::YardocTask.new
+
+  desc "Clean/remove the generated YARD Documentation cache"
+  task :clean do
+    rakefile_path = File.expand_path(File.dirname(__FILE__))
+    FileUtils.rm_rf(File.join(rakefile_path,YARD_DIR))
+  end
+
+  desc "Tell me about YARD undocummented objects"
+  YARD::Rake::YardocTask.new(:undoc) do |t|
+    t.stats_options = ["--list-undoc"]
+  end
+
+  desc "Measure YARD coverage"
+  require "yardstick/rake/measurement"
+  Yardstick::Rake::Measurement.new(:measure) do |measurement|
+    measurement.output = "yardstick/report.txt"
+  end
+
+  desc "Verify YARD coverage"
+  require "yardstick/rake/verify"
+  config = {"require_exact_threshold" => false}
+  Yardstick::Rake::Verify.new(:verify, config) do |verify|
+    verify.threshold = 57 # FIXME
+  end
+
+  desc "Generate static project architecture graph. (Calls docs:yard)"
+  # this calls `yard graph` so we can"t use the yardoc tasks like above
+  #   We could create a YARD:CLI:Graph object.
+  #   But we still have to send the output to the graphviz processor, etc.
+  task arch: [:yard] do
+    # rake can be run from any dir under here
+    #   so we don't know where we are
+    #   yard graph needs access to the lib file tree, and you can't specify it
+    #   to yard.  so we need to chdir here.
+    original_dir = Dir.pwd
+    Dir.chdir(File.expand_path(File.dirname(__FILE__)))
+    graph_processor = "dot"
+    if exe_exists?(graph_processor)
+      FileUtils.mkdir_p(DOCS_DIR)
+      if system("yard graph --full | #{graph_processor} -Tpng " \
+          "-o #{DOCS_DIR}/arch_graph.png")
+        puts "we made you a class diagram: #{DOCS_DIR}/arch_graph.png"
+      end
+    else
+      puts "ERROR: you don't have dot/graphviz; punting"
```

Comment:
Is it really an error if it doesn't cause a failure.  Should it be a warning?

Reply:
it fails to produce the image.  because it can't convert the dot format to png.
i could put that in stderr, if that helps, and exit 1